app/src/main/kotlin/com/wire/android/ui/userprofile/other/OtherUserProfileScreenViewModel.kt

```diff
@@ -210,6 +219,43 @@ class OtherUserProfileScreenViewModel @Inject constructor(
         _infoMessage.emit(type.uiText)
     }
 
+    fun openRemoveConversationMemberDialog() {
+        viewModelScope.launch {
+            removeConversationMemberDialogState = DialogState.Visible(
+                PreservedState.State(
+                    RemoveConversationMemberState(
+                        conversationId = conversationId!!,
+                        fullName = state.fullName,
+                        userName = state.userName,
+                        userId = userId
+                    )
+                )
+            )
+        }
+    }
+
+    fun hideRemoveConversationMemberDialog() {
+        viewModelScope.launch {
+            removeConversationMemberDialogState = DialogState.Hidden
+        }
+    }
+
+    fun removeConversationMember(preservedState: PreservedState<RemoveConversationMemberState>) {
+        viewModelScope.launch {
```

Comment:
Maybe we could switch the dispatcher to `IO` 🤔  so we offload main thread

Reply:
Should we wrap all use cases in this view model in dispatcher.io()?